src/subscription_manager/managerlib.py

```diff
@@ -856,7 +856,7 @@ def clean_all_data(backup=True):
     # for deleting persistent caches
     cache.ProfileManager.delete_cache()
     cache.InstalledProductsManager.delete_cache()
-    cache.SyspurposeCache.delete_cache()
+    # TODO Clear the syspurpose cache if unregistered.
```

Comment:
Going to implement this TODO now or later? What is the effect of not implementing now?

Reply:
Fixed